src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -707,23 +707,15 @@ trait TypeDiagnostics extends splain.SplainDiagnostics {
           )
           val why = if (sym.isPrivate) "private" else "local"
           var cond = "is never used"
-          val what = (
+          val what =
             if (sym.isDefaultGetter) "default argument"
             else if (sym.isConstructor) "constructor"
-            else if (
-              sym.isVar
-                || sym.isGetter && (sym.accessed.isVar || (sym.owner.isTrait && !sym.hasFlag(STABLE)))
-            ) s"var ${sym.name.getterName.decoded}"
```

Comment:
I guess it was trying to notice `x/x_=` pairs in traits? I would have to try it out.

Reply:
It looked redundant to me. `isVar` is already defined as:
```scala
def isVar: Boolean = isField && !isLazy && isMutableVal
def isMutableVal = if (owner.isTrait) !hasFlag(STABLE) else isMutable
```

Reply:
also I only read the first comment afterward.